src/framework/core/js/RemoteModel.js

```diff
@@ -0,0 +1,201 @@
+                value: false
+            }
+        },
+        invokers: {
+            fetch: {
+                funcName: "fluid.remoteModelComponent.fetch",
+                args: ["{that}"]
+            },
+            fetchImpl: "fluid.notImplemented",
+            write: {
+                funcName: "fluid.remoteModelComponent.write",
+                args: ["{that}"]
+            },
+            writeImpl: "fluid.notImplemented"
+        }
+    });
+
+    fluid.remoteModelComponent.launchPendingRequest = function (that) {
+        if (!that.model.requestInFlight) {
+            if (that.pendingRequests.fetch) {
+                that.fetch();
+            } else if (that.pendingRequests.write) {
+                that.write();
+            }
+        }
+    };
+
+    fluid.remoteModelComponent.updateModelFromFetch = function (that, fetchedModel) {
+        var remoteChanges = fluid.modelPairToChanges(fetchedModel, that.model.remote, "local");
+        var localChanges = fluid.modelPairToChanges(that.model.local, that.model.remote, "local");
+        var changes = remoteChanges.concat(localChanges);
+
+        // perform model updates in a single transaction
+        var transaction = that.applier.initiate();
+        transaction.fireChangeRequest({path: "local", type: "DELETE"}); // clear old local model
+        transaction.change("local", that.model.remote); // reset local model to the base for applying changes.
+        transaction.fireChangeRequest({path: "remote", type: "DELETE"}); // clear old remote model
+        transaction.change("remote", fetchedModel); // update remote model to fetched changes.
+        fluid.fireChanges(transaction, changes); // apply changes from remote and local onto base model.
+        transaction.commit(); // submit transaction
+    };
+
+    fluid.remoteModelComponent.makeSequneceStrategy = function (payload) {
+        return {
+            invokeNext: function (that) {
+                var lisrec = that.sources[that.index];
+                lisrec.listener = fluid.event.resolveListener(lisrec.listener);
+                var value = lisrec.listener.apply(null, [payload, that.options]);
+                return value;
+            },
+            resolveResult: function () {
+                return payload;
+            }
+        };
+    };
+
+    fluid.remoteModelComponent.makeSequence = function (listeners, payload, options) {
+        var sequencer = fluid.promise.makeSequencer(listeners, options, fluid.remoteModelComponent.makeSequneceStrategy(payload));
+        fluid.promise.resumeSequence(sequencer);
+        return sequencer;
+    };
+
+    fluid.remoteModelComponent.fireEventSequence = function (event, payload, options) {
+        var listeners = fluid.makeArray(event.sortedListeners);
+        var sequence = fluid.remoteModelComponent.makeSequence(listeners, payload, options);
+        return sequence.promise;
+    };
+
+    fluid.remoteModelComponent.fetch = function (that) {
```

Comment:
Substantial comments needed on both fetch and write - ideally we should write up the indented pseudocode we worked on and polish it a bit. Perhaps some kind of tree diagram/sequence diagram?

Reply:
@amb26 I've added comments to the fetch and write functions. I've created the following diagrams that can be added to the documentation that should be written for the `fluid.remoteModelComponent`.

![fetch diagram](https://user-images.githubusercontent.com/556024/34724612-70431dfc-f51c-11e7-9466-ee11a0fccf84.png)
![write diagram](https://user-images.githubusercontent.com/556024/34724617-733b55b0-f51c-11e7-8236-b9f7c5142f3f.png)